doc/api/vm.markdown

```diff
@@ -301,6 +301,38 @@ e.g. `(0,eval)('code')`. However, it also has the following additional options:
 - `timeout`: a number of milliseconds to execute `code` before terminating
   execution. If execution is terminated, an [`Error`][] will be thrown.
 
+## Example: Run a Server within a VM
+
+The context of `.runInThisContext()` refers to the v8 context. The code passed
+to this VM context will have it's own isolated scope. To run a simple web server
+using the `http` module, for instance, the code passed to the context must either
+call `require('http')` on it's own, or have a reference to the `http` module passed
+to it. For instance:
+
+```js
+'use strict'
+const vm = require('vm')
+
+let code = `
+  (function(require) {
+
+    const http = require('http');
+
+    http.createServer( (request, response) => {
+      response.writeHead(200, {'Content-Type': 'text/plain'});
+      response.end('Hello World\\n');
+    }).listen(8124);
+
+    console.log('Server running at http://127.0.0.1:8124/');
+  })
+`
+
+vm.runInThisContext(code)(require)
```

Comment:
This will not work, because then you'd likely need to pass all objects you want to use including `console`. Also goal for this example is not security, it's running node code. For more security you'd want to run this in a separate process.

```console
evalmachine.<anonymous>:11
    console.log('Server running at http://127.0.0.1:8124/');
    ^
ReferenceError: console is not defined
    at evalmachine.<anonymous>:11:5
    at Object.<anonymous> (/Users/jefe/repos/node_rename_2/runinnewcontext.js:18:25)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:146:18)
    at node.js:404:3
```